tests/test_notify_async.py

```diff
@@ -71,14 +71,16 @@ async def receiver():
         gen = aconn.notifies()
         async for n in gen:
             ns.append((n, time()))
-            if len(ns) >= 2:
-                await gen.aclose()
+            if len(ns) == 2:
+                await aconn.execute("notify foo, '3'")
+            if len(ns) == 3:
+                break
```

Comment:
I don't get this test. It doesn't seem to test the feature you introduced.

Can you demonstrate in a test that notifications generated in a concurrent thread or task are actually collected by the receiving generator?

Reply:
I have improved and expanded this test case now (also adding an additional case to test the behavior during a long-running query).